qirlib/src/tests.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{module, Builder};
+use inkwell::context::Context;
+use normalize_line_endings::normalized;
+use std::{env, fs, path::PathBuf};
+
+/// Compares generated IR against reference files in the "resources/tests" folder. If changes
+/// to code generation break the tests:
+///
+/// 1. Run the tests with the `PYQIR_TEST_SAVE_REFERENCES` environment variable set to
+///    regenerate the reference files.
+/// 2. Review the changes and make sure they look reasonable.
+/// 3. Unset the environment variable and run the tests again to confirm that they pass.
+pub(crate) fn assert_reference_ir(
```

Comment:
Should this use #[cfg(test)] so that is isn't shipped in the binary?

Reply:
This module in its entirety is `#[cfg(test)]` in lib.rs.